tests/test_cli.py

```diff
@@ -0,0 +1,145 @@
+    def test_correct_call_DownloadManager_without_file_prefix(
+            self, mock_downloadmethod, mock_downloadmanagerinit):
+        """shall return the proper call (with 'use_file_prefix': False) for the case of "
+        "lack of file prefix ('-p') in arguments"""
+        sys.argv.remove('-p')
+        sys.argv.remove('-r')
+        sys.argv[4] = '2019-10-22'
+        arg = cli_download.parse_args()
+        cli_download.main(arg)
+        self.assertIn({
+            'download_directory': '/test_folder/%Y_nh_polstere',
+            'geographic_location__geometry__intersects':
+            GEOSGeometry('POLYGON ((-22 84, -22 74, 32 74, 32 84, -22 84))'),
+            'max_downloads': 30600,
+            'provider_settings_path': '/config_folder/config_file.yml',
+            'time_coverage_end__lte': datetime(2020, 8, 22, 0, 0, tzinfo=tzutc()),
+            'time_coverage_start__gte': datetime(2019, 10, 22, 0, 0, tzinfo=tzutc()),
+            'dataseturi__uri__contains': 'osisaf',
+            'source__instrument__short_name__icontains': 'AMSR2',
+            'use_file_prefix': False
+        },
+            mock_downloadmanagerinit.call_args)
+
+    @mock.patch('geospaas_processing.downloaders.DownloadManager.__init__', return_value=None)
+    @mock.patch('geospaas_processing.downloaders.DownloadManager.download')
+    def test_correct_call_DownloadManager_with_file_prefix(
+            self, mock_downloadmethod, mock_downloadmanagerinit):
+        "shall return the proper call for the case of lack of two definite time points in arguments"
+        sys.argv.remove('-r')
+        sys.argv[4] = '2019-10-22'
+        arg = cli_download.parse_args()
+        cli_download.main(arg)
+        self.assertIn({
+            'download_directory': '/test_folder/%Y_nh_polstere',
+            'geographic_location__geometry__intersects':
+            GEOSGeometry('POLYGON ((-22 84, -22 74, 32 74, 32 84, -22 84))'),
+            'max_downloads': 30600,
+            'provider_settings_path': '/config_folder/config_file.yml',
+            'time_coverage_end__lte': datetime(2020, 8, 22, 0, 0, tzinfo=tzutc()),
+            'time_coverage_start__gte': datetime(2019, 10, 22, 0, 0, tzinfo=tzutc()),
+            'dataseturi__uri__contains': 'osisaf',
+            'source__instrument__short_name__icontains': 'AMSR2',
+            'use_file_prefix': True
+        },
+            mock_downloadmanagerinit.call_args)
+
+    @mock.patch('geospaas_processing.downloaders.DownloadManager.__init__', return_value=None)
+    @mock.patch('geospaas_processing.downloaders.DownloadManager.download')
+    def test_correct_call_DownloadManager_with_relative_time(
+            self, mock_downloadmethod, mock_downloadmanagerinit):
+        "shall return the proper call for the case of lack of relative time definition in arguments"
+        sys.argv[4] = "40"
+        arg = cli_download.parse_args()
+        with freeze_time("2012-01-14"):
+            cli_download.main(arg)
```

Comment:
Isn't installing and using  a new lib a little overkill in this case?
Couldn't you just mock `datetime.now()`?

Reply:
It is not possible to mock `datetime` because `datetime` is a built-in python datatype. We have to use that package in order to have a constant time for testing at different and various times!
It is not possible to mock `datetime.now()`.

Reply:
ok